activerecord/lib/active_record/test_fixtures.rb

```diff
@@ -106,7 +120,8 @@ def fixture(fixture_set_name, *fixture_names)
 
     private
       def run_in_transaction?
-        use_transactional_tests &&
+        has_explicit_config = database_transactions_config.any? { |_, enabled| enabled }
+        (use_transactional_tests || has_explicit_config) &&
           !self.class.uses_transaction?(name)
```

Comment:
I'd forgotten about `uses_transaction`. It's probably too complicated to worry about supporting a per-test-method/per-database matrix, though.